app/app/controls/BaseControl.js

```diff
@@ -235,4 +236,33 @@ export class BaseControl {
     isDragging() {
         return this.isDragging;
     }
+
+    /**
+     * Check if control is within the boundaries of given rectangle.
+     * @param x1
+     * @param y1
+     * @param x2
+     * @param y2
+     * @returns {boolean}
+     */
+    isWithinArea(x1, y1, x2, y2) {
+        var x = this.componentBodyShape.getBBox().x;
```

Comment:
Thank you for comment.

Implementing it for every subclass will make it hard to maintain code. Are you sure?

I did it in my local repo, created new additional branch for this commit. In case, you will want it, I'll cherry-pick this commit to selection branch or merge two branches.